nav2_graceful_controller/src/graceful_controller.cpp

```diff
@@ -243,58 +284,66 @@ void GracefulController::setSpeedLimit(
   }
 }
 
-geometry_msgs::msg::PoseStamped GracefulController::getMotionTarget(
-  const double & motion_target_dist,
-  const nav_msgs::msg::Path & transformed_plan)
-{
-  // Find the first pose which is at a distance greater than the motion target distance
-  auto goal_pose_it = std::find_if(
-    transformed_plan.poses.begin(), transformed_plan.poses.end(), [&](const auto & ps) {
-      return std::hypot(ps.pose.position.x, ps.pose.position.y) >= motion_target_dist;
-    });
-
-  // If the pose is not far enough, take the last pose
-  if (goal_pose_it == transformed_plan.poses.end()) {
-    goal_pose_it = std::prev(transformed_plan.poses.end());
-  }
-
-  return *goal_pose_it;
-}
-
 bool GracefulController::simulateTrajectory(
-  const geometry_msgs::msg::PoseStamped & robot_pose,
   const geometry_msgs::msg::PoseStamped & motion_target,
   const geometry_msgs::msg::TransformStamped & costmap_transform,
-  nav_msgs::msg::Path & trajectory, const bool & backward)
+  nav_msgs::msg::Path & trajectory,
+  geometry_msgs::msg::TwistStamped & cmd_vel,
+  bool backward)
 {
-  // Check for collision before moving
-  if (inCollision(
-      robot_pose.pose.position.x, robot_pose.pose.position.y,
-      tf2::getYaw(robot_pose.pose.orientation)))
-  {
-    return false;
-  }
+  trajectory.poses.clear();
 
-  // First pose
+  // First pose is robot current pose
   geometry_msgs::msg::PoseStamped next_pose;
   next_pose.header.frame_id = costmap_ros_->getBaseFrameID();
   next_pose.pose.orientation.w = 1.0;
-  trajectory.poses.push_back(next_pose);
+
+  // Should we simulate rotation initially?
+  bool sim_initial_rotation = has_new_path_ && (params_->initial_rotation_tolerance > 0.0);
```

Comment:
I think there should be an explicit enable `bool` rather than checking if the `tol > 0`

Reply:
We disagree on this, but I will switch it back to that style

Reply:
Or I suppose we could keep it as that as long as the docs make it clear that if its set to `0.0` or ` < 0.0` it'll be interpreted as disabled. We have _some_ parameters like that around. I'd like add in the `-1` case as also disabled too

Reply:
I reverted to two parameters - which also seems to help the diff look smaller since git is less confused